# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [ ]:
a = 101
b = 210
c = 111
d = 402

print (((a + b) * c)/d)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [ ]:
 a = 11283
 b = 89
 c = a % b
 print (c)

69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [ ]:
a = 11283
b = 89
c = a // b
print (c)

126


4. Запишіть у змінну `name` ваше імʼя.

In [ ]:
name = 'Iryna'

5. Запишіть у змінну `surname` ваше прізвище.

In [37]:
surname = 'Danchenko'

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [ ]:
name = 'Iryna'
surname = 'Danchenko'
name_surname = name + " " + surname
print (name_surname)

Iryna Danchenko


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [ ]:
name_surname = "Iryna Danchenko"
print (name_surname. upper ())

IRYNA DANCHENKO


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [ ]:
name_surname = "Iryna Danchenko"
name_surname [2]

'y'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [ ]:
surname = "Danchenko"
print (surname [0], surname [8])

D o


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [ ]:
surname = "Danchenko"
len (surname)

9

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [ ]:
capital = 'Antananarivo'
count_a = capital. lower (). count ('a')
print(count_a)

4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [41]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
starts = my_string. startswith ('The')
ends = my_string. endswith ('dog!')
print (starts)
print (ends)

True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [47]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
words = my_string. split ()
print (words)
joined = '_'.join (words)
print (joined)

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']
The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`